stdlib/public/core/UnsafePointer.swift

```diff
@@ -444,16 +444,14 @@ extension UnsafePointer {
   /// - Returns: A pointer to the stored property represented
   ///            by the key path, or `nil`.
   @_alwaysEmitIntoClient
-  #if $Embedded
   @_transparent
-  #endif
   public func pointer<Property>(
     to property: KeyPath<Pointee, Property>
   ) -> UnsafePointer<Property>? {
     guard let o = property._storedInlineOffset else { return nil }
     _internalInvariant(o >= 0)
     _debugPrecondition(
-      o == 0 || UnsafeRawPointer(self) < UnsafeRawPointer(bitPattern: 0 &- o)!,
+      o == 0 || UInt(bitPattern: self) < UInt(bitPattern: 0 &- o),
```

Comment:
What is this debug precondition actually checking? This is always satisfied if `UnsafePointer` is a valid pointer to `Pointee`, right? We don't normally check pointer arithmetic on UP.

That aside, I think this can probably be written more clearly as something like: `!UInt(bitPattern: self).addingReportingOverflow(UInt(bitPattern: o)).overflow`

Reply:
That's exactly what this precondition is checking. I didn't think of going the `UInt` route originally; the pointer arithmetic is clearly worse.

Reply:
Compare the debug codegen for an extracted example, using addingReportingOverflow trims the instruction count by about 1/3 (it's still absurdly expensive, however): https://godbolt.org/z/TGTY7M1Wb

Reply:
(This is precisely the sort of operation where we really want some sort of "inline-always-but-only-if-fully-specialized" annotation; we really want this outlined in debug)

Reply:
👍 nice! I changed it.